src/libsyntax/str.rs

```diff
@@ -8,6 +8,18 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// FIXME: This was copied from core/str/mod.rs because it is currently unstable.
 pub fn char_at(s: &str, byte: usize) -> char {
     s[byte..].chars().next().unwrap()
 }
+
+// FIXME: This was copied from core/str/mod.rs because it is currently unstable.
+#[inline]
+pub fn slice_shift_char(s: &str) -> Option<(char, &str)> {
```

Comment:
It looks like this function is actually pretty rarely used (only used twice here), could this just be removed entirely and each call site just have a custom implementation? They don't really need `slice_shift_char` in its full glory but rather just some smaller specialized versions.